backend/src/main/java/click/snekhome/backend/service/NodeService.java

```diff
@@ -39,4 +42,67 @@ public Node add(NodeData nodeData) {
         return node;
     }
 
+    public Node edit(String id, ActionType actionType) {
+        Node node = this.nodeRepo.findById(id).orElseThrow();
+        Node newNode;
+        if (node.ownerId() == null) {
+            if (actionType == ActionType.HACK) {
+                newNode = new Node(
+                        node.id(),
+                        PLAYERNAME,
+                        node.name(),
+                        node.level() + 1,
+                        100,
+                        node.coordinates(),
+                        Instant.now().getEpochSecond(),
+                        node.lastAttack()
+                );
+                return this.nodeRepo.save(newNode);
+            }
+        } else {
+            if (actionType == ActionType.HACK) {
+                newNode = new Node(
+                        node.id(),
+                        node.ownerId(),
+                        node.name(),
+                        node.level() + 1,
+                        100,
+                        node.coordinates(),
+                        Instant.now().getEpochSecond(),
+                        node.lastAttack()
+                );
+                return this.nodeRepo.save(newNode);
+            } else if (actionType == ActionType.ABANDON) {
```

Comment:
what happen in the third case ?

Reply:
I want to make sure that if a node isn't owned, a player can only `HACK` and therefore claim ownership of the node. 
If the node is owned by the player, the player can `HACK` it to increase the level or `ABANDON` it to decrease the level.
The logic for `HACK`ing a node that is owned by another player will be implemented later. 
You're right, I should clarify the `if ... else if ... else` further to make it more understandable. Thank you for the hint💪